### ISSUE TITLE
Typehinting entity fields doesnt work due to overzealous `EntityFieldsViaMagicReflectionExtension`

### ISSUE THREAD

**Issue by dpi** [mglaman/phpstan-drupal#172]
In an attempt to suggest types to PHPStan, I found adding a `@property` hint to the class did not work when `phpstan-drupal` 's `PHPStan\Reflection\EntityFieldsViaMagicReflectionExtension` was enabled in extension.neon.

```php
/**
 * ....
 *
 * @property \Drupal\Core\Field\EntityReferenceFieldItemList $myField
 */
class MyEntity extends EntityBase {

  /**
   * ...
   */
  public function getParent(): array {
    return $this->myField->referencedEntities();
  }
```

Above code results in:

> Call to an undefined method Drupal\Core\Field\FieldItemListInterface::referencedEntities().

PHPStan thinks `myField` is a `FieldItemListInterface`. If I disable phpstan-drupal PHPStan does not complain.

A `@property` should probably take priority.

**Comment by mglaman** [comment]
Oh, interesting. That'll be fun to figure out :)
